src/inference/src/ie_core.cpp

```diff
@@ -505,12 +518,156 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
         return res;
     }
 
+    void ApplyAutoBatching(const ie::CNNNetwork& network,
+                           std::string& deviceName,
+                           std::map<std::string, std::string>& config_with_batch) {
+        std::string deviceNameWithBatchSize, deviceNameWithoutBatch;
+        if (deviceName.find("BATCH") != std::string::npos) {
+            // explicitly enabled Auto-Batching
+            auto pos = deviceName.find_first_of(":");
+            if (pos != std::string::npos) {
+                deviceNameWithBatchSize = deviceName.substr(pos + 1);
+                deviceNameWithoutBatch = DeviceIDParser::getBatchDevice(deviceNameWithBatchSize);
+            }
+        } else {
+            // check whether  the Auto-Batching is disabled explicitly
+            const auto& batch_mode = config_with_batch.find(CONFIG_KEY(ALLOW_AUTO_BATCHING));
```

Comment:
what if `ALLOW_AUTO_BATCHING` is set globally as `core.set_config({ ALLOW_AUTO_BATCHING, OFF })`? I suppose it does not work currently.
See how it's implemented for `CACHE_DIR`

Reply:
will be fixed later as part of the CVS-74713